src/saltext/vmware/states/roles.py

```diff
@@ -0,0 +1,161 @@
+                        privileges:
+                        - Configure commands
+                        - Create
+                        - Remove
+                        - Modify
+                        - Recovery
+                    - group: Protection Group
+                        privileges:
+                        - Assign to plan
+                        - Create
+                        - Modify
+                        - Remove
+                        - Remove from plan
+    """
+
+    service_instance = service_instance or connect.get_service_instance(
+        config=__opts__, profile=profile
+    )
+    ret = {"name": name, "changes": {}, "result": None, "comment": ""}
+
+    # # Clone config input to a Map.
+    # # Can be used to transform input to internal objects and do validation if needed
+    new_config = {}
+    for role_config in config:
+        # Create full representation of the object, default or empty values
+        new_role_config = {"privileges": {}}
+        # Transform / Validate input vs object, e.g. constraints section
+        if "groups" in role_config:
+            for privilege in role_config["groups"]:
+                priv_group = privilege["group"]
+                new_role_config["privileges"][priv_group] = []
+                for item in privilege["privileges"]:
+                    new_role_config["privileges"][priv_group].append(item)
+        new_config[role_config["role"]] = new_role_config
+
+    log.debug("---------------NEW--------------")
+    log.debug(json.dumps(new_config, indent=2))
+    log.debug("---------------NEW--------------")
+
+    # Get all policies from vCenter, the objects are VMOMI objects
+    old_configs = roles_module.find(
+        role_name=None, service_instance=service_instance, profile=profile
+    )
+
+    # old_configs must have only the rules that are in the scope of interest (provided in argument config_input)
+    old_config = {}
+    for role in old_configs:
+        if role["role"] not in new_config.keys():
+            continue
+
+        old_config[role["role"]] = {"privileges": role["privileges"]}
+
+    log.debug("--------------OLD---------------")
+    log.debug(json.dumps(old_config, indent=2))
+    log.debug("--------------OLD---------------")
+
+    # Find rules changes
+    changes = []
+    diffs = drift.drift_report(old_config, new_config, diff_level=0)
+    diffs = json.loads(json.dumps(diffs))  # clone object
+    if diffs is not None:
+        ret["changes"] = diffs
+
+        log.debug("==============DRIFT===============")
+        log.debug(json.dumps(diffs, indent=2))
+        log.debug("==============DRIFT===============")
+
+        # add changes for process if not dry-run
+        for d_name in diffs:
+            new_policy = diffs[d_name]["new"]
+            changes.append({**{"role": d_name}, **new_policy})
+
+    # If it's not dry-run and has changes, then apply changes
+    if not __opts__["test"] and changes:
+        success = True
```

Comment:
**suggestion** success should only be None if test=True, regardless of changes. This could pretty easily be:

```suggestion
    if __opts__["test"]:
        success = None
    elif changes:
        success = True
```